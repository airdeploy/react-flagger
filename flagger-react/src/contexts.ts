import {IEntity, IFlaggerConfiguration} from 'flagger/types/Types'
import React from 'react'

export interface IFlaggerCtx {
  entity?: IEntity
  config?: IFlaggerConfiguration
  loading: boolean
  getVariation: (codename: string, entity?: IEntity) => string
  getFlagDetails: (
    codename: string,
    entity?: IEntity
  ) => {variation: string; enabled: boolean; isSampled: boolean; payload: any}
}

export const flaggerCtx = React.createContext<IFlaggerCtx>({
  loading: false,
  getVariation: () => 'off',
  getFlagDetails: () => ({
    variation: 'off',
    enabled: false,
    isSampled: false,
    payload: null,
  }),
})
