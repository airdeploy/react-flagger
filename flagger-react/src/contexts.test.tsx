import React from 'react'
import renderer from 'react-test-renderer'

import {flaggerCtx} from './contexts'

describe('contexts.ts tests', () => {
  beforeAll(async () => {
    jest.mock('flagger')
  })

  afterAll(async () => {
    jest.clearAllMocks()
  })

  describe('flaggerCtx tests', () => {
    test('provides default value', () => {
      const component = renderer.create(
        <flaggerCtx.Consumer>
          {({loading, getFlagDetails, getVariation, config, entity}) => (
            <ul>
              <li>loading: {loading}</li>
              <li>
                getVariation:{' '}
                {JSON.stringify(getVariation('any-flag-name'), null, 2)}
              </li>
              <li>
                flagDetails:{' '}
                {JSON.stringify(getFlagDetails('any-flag-name'), null, 2)}
              </li>
              <li>entity: {JSON.stringify(entity, null, 2)}</li>
              <li>config: {JSON.stringify(config, null, 2)}</li>
            </ul>
          )}
        </flaggerCtx.Consumer>
      )

      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
