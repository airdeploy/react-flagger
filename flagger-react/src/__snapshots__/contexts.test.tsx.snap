// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contexts.ts tests flaggerCtx tests provides default value 1`] = `
<ul>
  <li>
    loading: 
  </li>
  <li>
    getVariation:
     
    "off"
  </li>
  <li>
    flagDetails:
     
    {
  "variation": "off",
  "enabled": false,
  "isSampled": false,
  "payload": null
}
  </li>
  <li>
    entity: 
  </li>
  <li>
    config: 
  </li>
</ul>
`;
